server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/utils/StarTreeDocumentsSorterTests.java

```diff
@@ -24,46 +26,87 @@
  * Tests for {@link StarTreeDocumentsSorter}.
  */
 public class StarTreeDocumentsSorterTests extends OpenSearchTestCase {
```

Comment:
Lets add tests in flush/merge UTs and also in integ tests. Also how does unsigned long work as a metric ? Can we test that as well ?

Reply:
1. Can we also please test the behavior of metrics for unsigned long ? And tests around the same - the results should be same as normal aggregation query on unsigned long fields.

2. We need to add tests to mapper as well. You can refer to https://github.com/opensearch-project/OpenSearch/pull/16641

Reply:
Done.